### PR TITLE
 Added Custom Pooja option and Amount paid option

### PR DESCRIPTION
**Pull request by gopikaharikumar**
Fixes issue #11 
- Added an option for Custom Pooja in the App
<table>
<tr>
<td></td>
<td></td>
</tr>
</table>

- Added an option to enter the Amount paid for a Pooja
<table>
<tr>
<td></td>
<td></td>
</tr
</table>

### CODE REVIEW

File: TempleApp/src/main/java/example/TempleApp/Activities/InsertData.java [hunk 1]
```diff
@@ -24,11 +25,13 @@
 
 public class InsertData extends AppCompatActivity {
 
-    String id,name,poojaTyp,overall,money,paidCheck ="NOT PAID";
+    String id,name,poojaTyp,overall,money,paidCheck ="NOT PAID", Amount;
```
Comment (yashk2000) [inline TempleApp/src/main/java/example/TempleApp/Activities/InsertData.java]: Avoid starting variable names with capital letters. Please give it a different name starting with a lowercase character.

File: TempleApp/src/main/java/example/TempleApp/Activities/InsertData.java [hunk 2]
```diff
@@ -92,15 +118,27 @@ public void onClick(View v) {
...
-                        poojaTyp = String.valueOf(poojaType.getSelectedItem());
+                        if (sp.contentEquals("Custom Pooja")) {
+                            String name= custompuja.getText().toString();
+                            poojaTyp= name;
+                            Amount= amount.getText().toString();
+                        }
+                        else {
+                            poojaTyp = String.valueOf(poojaType.getSelectedItem());
+                            Amount= amount.getText().toString();
+
```
Comment (yashk2000) [inline TempleApp/src/main/java/example/TempleApp/Activities/InsertData.java]: Remove the extra line.

File: TempleApp/src/main/java/example/TempleApp/Activities/InsertData.java [hunk 3]
```diff
@@ -92,15 +118,27 @@ public void onClick(View v) {
...
+                        }
+
```
Comment (yashk2000) [inline TempleApp/src/main/java/example/TempleApp/Activities/InsertData.java]: Extra line. Please remove.

File: TempleApp/src/main/java/example/TempleApp/Activities/ReadSingleData.java [hunk 4]
```diff
@@ -63,12 +64,12 @@ public void onCheckedChanged(RadioGroup group, int checkedId) {
                     case R.id.radio_donate:
                         totalLayout.setVisibility(View.VISIBLE);
                         flag = 0;
-                        Toast.makeText(getBaseContext(), getString(R.string.selected_donate), Toast.LENGTH_LONG).show();
+                        Toast.makeText(getBaseContext(), "Selected To Donate Money", Toast.LENGTH_LONG).show();
```
Comment (yashk2000) [inline TempleApp/src/main/java/example/TempleApp/Activities/ReadSingleData.java]: Any particular reason for replacing the string resource with hardcoded string? If not please revert back to using string resources.

File: TempleApp/src/main/java/example/TempleApp/Activities/ReadSingleData.java [hunk 5]
```diff
@@ -63,12 +64,12 @@ public void onCheckedChanged(RadioGroup group, int checkedId) {
...
                         break;
                     case R.id.radio_pooja:
                         totalLayout.setVisibility(View.VISIBLE);
                         flag = 1;
-                        Toast.makeText(getBaseContext(), getString(R.string.selected_register), Toast.LENGTH_LONG).show();
+                        Toast.makeText(getBaseContext(), "Selected To Register New Pooja", Toast.LENGTH_LONG).show();
```
Comment (yashk2000) [inline TempleApp/src/main/java/example/TempleApp/Activities/ReadSingleData.java]: String resources.

File: TempleApp/src/main/java/example/TempleApp/Activities/ReadSingleData.java [hunk 6]
```diff
@@ -80,9 +81,9 @@ public void onCheckedChanged(RadioGroup group, int checkedId) {
             public void onClick(View view) {
 
                 if (flag == 1) {
-                    id = getString(R.string.REG) + uid1ET.getText().toString();
+                    id = "REG" + uid1ET.getText().toString();
```
Comment (yashk2000) [inline TempleApp/src/main/java/example/TempleApp/Activities/ReadSingleData.java]: String resources.

File: TempleApp/src/main/java/example/TempleApp/Activities/ReadSingleData.java [hunk 7]
```diff
@@ -80,9 +81,9 @@ public void onCheckedChanged(RadioGroup group, int checkedId) {
...
                 } else {
-                    id = getString(R.string.DON) + uid1ET.getText().toString();
+                    id = "DON" + uid1ET.getText().toString();
```
Comment (yashk2000) [inline TempleApp/src/main/java/example/TempleApp/Activities/ReadSingleData.java]: String resources.

File: TempleApp/src/main/java/example/TempleApp/Activities/UpdateData.java [hunk 8]
```diff
@@ -24,14 +25,16 @@
 
 public class UpdateData extends AppCompatActivity {
 
-    String id;
+    String id, Amount;
```
Comment (yashk2000) [inline TempleApp/src/main/java/example/TempleApp/Activities/UpdateData.java]: Rename the variable such that it starts with a lower case character.

File: TempleApp/src/main/java/example/TempleApp/Activities/UpdateData.java [hunk 9]
```diff
@@ -78,6 +85,27 @@ public void onCheckedChanged(RadioGroup group, int checkedId) {
...
+            public void onItemSelected(AdapterView<?> adapterView, View view, int i, long l) {
+
+                if (i == 4) {
+                    custompuja.setVisibility(View.VISIBLE);
+                    amount.setVisibility(View.VISIBLE);
+                } else {
+                    custompuja.setVisibility(View.GONE);
+                    amount.setVisibility(View.VISIBLE);
+                }
+
```
Comment (yashk2000) [inline TempleApp/src/main/java/example/TempleApp/Activities/UpdateData.java]: Extra line.

File: TempleApp/src/main/java/example/TempleApp/Activities/UpdateData.java [hunk 10]
```diff
@@ -101,12 +129,24 @@ public void onClick(View v) {
...
-                    poojaTyp = String.valueOf(poojaType.getSelectedItem());
+                    if (sp.contentEquals("Custom Pooja")) {
+                        String name= custompuja.getText().toString();
+                        poojaTyp= name;
+                        Amount= amount.getText().toString();
+                    }
+                    else {
+                        poojaTyp = String.valueOf(poojaType.getSelectedItem());
+                        Amount= amount.getText().toString();
+
```
Comment (yashk2000) [inline TempleApp/src/main/java/example/TempleApp/Activities/UpdateData.java]: Extra line.

File: TempleApp/src/main/java/example/TempleApp/Activities/UpdateData.java [hunk 11]
```diff
@@ -101,12 +129,24 @@ public void onClick(View v) {
...
+                    }
+
```
Comment (yashk2000) [inline TempleApp/src/main/java/example/TempleApp/Activities/UpdateData.java]: Extra line.

File: TempleApp/src/main/res/layout/insert_data.xml [hunk 12]
```diff
@@ -63,6 +63,8 @@
                 android:text="@string/select_pooja_type"
                 android:textSize="15sp" />
 
+
```
Comment (yashk2000) [inline TempleApp/src/main/res/layout/insert_data.xml]: Remove both the extra lines here.

File: TempleApp/src/main/res/layout/insert_data.xml [hunk 13]
```diff
@@ -80,13 +82,29 @@
...
                 android:layout_margin="45dp"
                 android:hint="@string/name" />
 
+            <EditText
+                android:id="@+id/amount"
+                android:layout_width="match_parent"
+                android:layout_height="50dp"
+                android:layout_margin="45dp"
+                android:hint="amount" />
+
```
Comment (yashk2000) [inline TempleApp/src/main/res/layout/insert_data.xml]: Extra lines.

File: TempleApp/src/main/res/layout/read_data.xml [hunk 14]
```diff
@@ -128,6 +128,16 @@
...
 
+        <TextView
+            android:id="@+id/am"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_below="@id/paid_l"
+            android:layout_alignStart="@id/paid_l"
+            android:layout_alignLeft="@id/paid_l"
+            android:layout_marginTop="35dp" />
+
```
Comment (yashk2000) [inline TempleApp/src/main/res/layout/read_data.xml]: Extra line.

**Comment by yashk2000** [comment]
@gopikaharikumar also please follow the pr template properly. Mention the changes you have made and add screenshots of any UI/UX changes made.

File: TempleApp/src/main/java/example/TempleApp/Activities/InsertData.java [hunk 15]
```diff
@@ -70,6 +77,25 @@ public void onCheckedChanged(RadioGroup group, int checkedId) {
...
+            public void onItemSelected(AdapterView<?> adapterView, View view, int i, long l) {
+
+                if (i == 4) {
+                    custompuja.setVisibility(View.VISIBLE);
+                    amount.setVisibility(View.VISIBLE);
+                } else {
+                    custompuja.setVisibility(View.GONE);
+                    amount.setVisibility(View.VISIBLE);
+                }
+
```
Comment (yashk2000) [inline TempleApp/src/main/java/example/TempleApp/Activities/InsertData.java]: Extra line

File: TempleApp/src/main/java/example/TempleApp/Activities/ReadSingleData.java [hunk 16]
```diff
@@ -141,30 +142,33 @@ protected void onPostExecute(Void aVoid) {
...
+                String[] str = name.split(getResources().getString(R.string.empty));
+                name_v.setText(str[2]);
+                am.setText("Amount:    Rs."+str[1]);
                 pooja_l.setText(str[0]);
-                paid_v.setText(str[2]);
+                paid_v.setText(str[3]);
                 if (flag == 0) {
-                    pooja_v.setText(getString(R.string.type_of_pooja));
+                    pooja_v.setText("Type of pooja: ");
+
```
Comment (yashk2000) [inline TempleApp/src/main/java/example/TempleApp/Activities/ReadSingleData.java]: extra

File: TempleApp/src/main/java/example/TempleApp/Activities/ReadSingleData.java [hunk 17]
```diff
@@ -141,30 +142,33 @@ protected void onPostExecute(Void aVoid) {
...
+
```
Comment (yashk2000) [inline TempleApp/src/main/java/example/TempleApp/Activities/ReadSingleData.java]: extra

File: TempleApp/src/main/java/example/TempleApp/Activities/ReadSingleData.java [hunk 18]
```diff
@@ -101,27 +102,27 @@ protected void onPreExecute() {
             super.onPreExecute();
 
             dialog = new ProgressDialog(ReadSingleData.this);
-            dialog.setTitle(getString(R.string.wait));
-            dialog.setMessage(getString(R.string.fetch));
+            dialog.setTitle("Hey Wait Please...");
```
Comment (yashk2000) [inline TempleApp/src/main/java/example/TempleApp/Activities/ReadSingleData.java]: String resource to be used.

File: TempleApp/src/main/java/example/TempleApp/Activities/ReadSingleData.java [hunk 19]
```diff
@@ -101,27 +102,27 @@ protected void onPreExecute() {
...
+            dialog.setMessage("Fetching Your Values");
```
Comment (yashk2000) [inline TempleApp/src/main/java/example/TempleApp/Activities/ReadSingleData.java]: String resource to be used.

File: TempleApp/src/main/java/example/TempleApp/Activities/ReadSingleData.java [hunk 20]
```diff
@@ -101,27 +102,27 @@ protected void onPreExecute() {
...
             dialog.show();
 
         }
 
         @Nullable
         @Override
         protected Void doInBackground(Void... params) {
-            Log.i(Controller.TAG, getString(R.string.ID_value) + id);
+            Log.i(Controller.TAG, "IDVALUE" + id);
```
Comment (yashk2000) [inline TempleApp/src/main/java/example/TempleApp/Activities/ReadSingleData.java]: String resource to be used.

File: TempleApp/src/main/java/example/TempleApp/Activities/ReadSingleData.java [hunk 21]
```diff
@@ -101,27 +102,27 @@ protected void onPreExecute() {
...
             JSONObject jsonObject = Controller.readData(id);
-            Log.i(Controller.TAG, getString(R.string.json_obj) + jsonObject);
+            Log.i(Controller.TAG, "Json obj " + jsonObject);
```
Comment (yashk2000) [inline TempleApp/src/main/java/example/TempleApp/Activities/ReadSingleData.java]: String resource to be used.

File: TempleApp/src/main/java/example/TempleApp/Activities/ReadSingleData.java [hunk 22]
```diff
@@ -101,27 +102,27 @@ protected void onPreExecute() {
...
 
             try {
                 /*
                   Check Whether Its NULL???
                  */
                 if (jsonObject != null) {
 
-                    JSONObject user = jsonObject.getJSONObject(getString(R.string.user));
-                    name = user.getString(getString(R.string.name_));
+                    JSONObject user = jsonObject.getJSONObject("user");
```
Comment (yashk2000) [inline TempleApp/src/main/java/example/TempleApp/Activities/ReadSingleData.java]: String resource to be used.

File: TempleApp/src/main/java/example/TempleApp/Activities/ReadSingleData.java [hunk 23]
```diff
@@ -101,27 +102,27 @@ protected void onPreExecute() {
...
+                    name = user.getString("name");
```
Comment (yashk2000) [inline TempleApp/src/main/java/example/TempleApp/Activities/ReadSingleData.java]: String resource to be used.

File: TempleApp/src/main/java/example/TempleApp/Activities/ReadSingleData.java [hunk 24]
```diff
@@ -141,30 +142,33 @@ protected void onPostExecute(Void aVoid) {
             }
 
             if (name != null) {
-                id_l.setText(getString(R.string.ID));
-                name_l.setText(getString(R.string.name));
+                id_l.setText("ID: ");
```
Comment (yashk2000) [inline TempleApp/src/main/java/example/TempleApp/Activities/ReadSingleData.java]: String resource to be used.

File: TempleApp/src/main/java/example/TempleApp/Activities/ReadSingleData.java [hunk 25]
```diff
@@ -141,30 +142,33 @@ protected void onPostExecute(Void aVoid) {
...
+                name_l.setText("Name: ");
```
Comment (yashk2000) [inline TempleApp/src/main/java/example/TempleApp/Activities/ReadSingleData.java]: String resource to be used.

File: TempleApp/src/main/java/example/TempleApp/Activities/ReadSingleData.java [hunk 26]
```diff
@@ -141,30 +142,33 @@ protected void onPostExecute(Void aVoid) {
...
                 int flag = 0;
-                if (id.substring(0, 3).equals(getString(R.string.DON))) {
+                if (id.substring(0, 3).equals("DON")) {
```
Comment (yashk2000) [inline TempleApp/src/main/java/example/TempleApp/Activities/ReadSingleData.java]: String resource to be used.

File: TempleApp/src/main/java/example/TempleApp/Activities/ReadSingleData.java [hunk 27]
```diff
@@ -141,30 +142,33 @@ protected void onPostExecute(Void aVoid) {
...
                     id_v.setText(id.substring(3, id.length()));
                     flag = 1;
-                } else if (id.substring(0, 3).equals(getString(R.string.REG))) {
+                } else if (id.substring(0, 3).equals("REG")) {
```
Comment (yashk2000) [inline TempleApp/src/main/java/example/TempleApp/Activities/ReadSingleData.java]: String resource to be used.

File: TempleApp/src/main/java/example/TempleApp/Activities/ReadSingleData.java [hunk 28]
```diff
@@ -141,30 +142,33 @@ protected void onPostExecute(Void aVoid) {
...
-                name_v.setText(str[1]);
+                String[] str = name.split(getResources().getString(R.string.empty));
+                name_v.setText(str[2]);
+                am.setText("Amount:    Rs."+str[1]);
                 pooja_l.setText(str[0]);
-                paid_v.setText(str[2]);
+                paid_v.setText(str[3]);
                 if (flag == 0) {
-                    pooja_v.setText(getString(R.string.type_of_pooja));
+                    pooja_v.setText("Type of pooja: ");
```
Comment (yashk2000) [inline TempleApp/src/main/java/example/TempleApp/Activities/ReadSingleData.java]: String resource to be used.

File: TempleApp/src/main/java/example/TempleApp/Activities/ReadSingleData.java [hunk 29]
```diff
@@ -141,30 +142,33 @@ protected void onPostExecute(Void aVoid) {
...
+
+
                 } else {
-                    pooja_v.setText(getString(R.string.money_donated));
+                    pooja_v.setText("Money Donated: ");
```
Comment (yashk2000) [inline TempleApp/src/main/java/example/TempleApp/Activities/ReadSingleData.java]: String resource to be used.

File: TempleApp/src/main/java/example/TempleApp/Activities/ReadSingleData.java [hunk 30]
```diff
@@ -141,30 +142,33 @@ protected void onPostExecute(Void aVoid) {
...
                 }
-                paid_l.setText(getString(R.string.paid_status));
+                paid_l.setText("Paid Status: ");
```
Comment (yashk2000) [inline TempleApp/src/main/java/example/TempleApp/Activities/ReadSingleData.java]: String resource to be used.

File: TempleApp/src/main/java/example/TempleApp/Activities/ReadSingleData.java [hunk 31]
```diff
@@ -141,30 +142,33 @@ protected void onPostExecute(Void aVoid) {
...
 
             } else
-                Toast.makeText(getApplicationContext(), getString(R.string.id_not_found), Toast.LENGTH_LONG).show();
+                Toast.makeText(getApplicationContext(), "ID not found", Toast.LENGTH_LONG).show();
```
Comment (yashk2000) [inline TempleApp/src/main/java/example/TempleApp/Activities/ReadSingleData.java]: String resource to be used.

**Comment by immadisairaj** [comment]
@gopikaharikumar what is the status?

File: TempleApp/src/main/java/example/TempleApp/activities/InsertData.java [hunk 32]
```diff
@@ -24,11 +25,13 @@
 
 public class InsertData extends AppCompatActivity {
 
-    String id,name,poojaTyp,overall,money,paidCheck ="NOT PAID";
+    String id,name,poojaTyp,overall,money,paidCheck ="NOT PAID", amnt;
     Spinner poojaType;
     RadioGroup radioGroup;
     LinearLayout totalLayout;
     EditText moneyDonated;
+    EditText custompuja;
```
Comment (immadisairaj) [inline TempleApp/src/main/java/example/TempleApp/activities/InsertData.java]: we should follow either "pooja" or "puja"
but "pooja" is used in the entire app it's better to use that.

File: TempleApp/src/main/java/example/TempleApp/activities/InsertData.java [hunk 33]
```diff
@@ -70,6 +77,24 @@ public void onCheckedChanged(RadioGroup group, int checkedId) {
...
+                    amount.setVisibility(View.VISIBLE);
+                } else {
+                    custompuja.setVisibility(View.GONE);
+                    amount.setVisibility(View.VISIBLE);
+                }
+            }
+
+            @Override
+            public void onNothingSelected(AdapterView<?> adapterView) {
+
```
Comment (immadisairaj) [inline TempleApp/src/main/java/example/TempleApp/activities/InsertData.java]: You can add some comment or log

File: TempleApp/src/main/res/values/strings.xml [hunk 34]
```diff
@@ -1,4 +1,4 @@
-<resources>
+ <resources>
```
Comment (immadisairaj) [inline TempleApp/src/main/res/values/strings.xml]: revert

File: TempleApp/src/main/java/example/TempleApp/activities/UpdateData.java [hunk 35]
```diff
@@ -70,16 +74,31 @@ public void onCheckedChanged(RadioGroup group, int checkedId) {
...
+                if (i == 4) {
+                    custompuja.setVisibility(View.VISIBLE);
+                    amount.setVisibility(View.VISIBLE);
+                } else {
+                    custompuja.setVisibility(View.GONE);
+                    amount.setVisibility(View.VISIBLE);
+                }
+            }
+            @Override
+            public void onNothingSelected(AdapterView<?> adapterView) {
```
Comment (immadisairaj) [inline TempleApp/src/main/java/example/TempleApp/activities/UpdateData.java]: same as above

File: TempleApp/src/main/java/example/TempleApp/activities/UpdateData.java [hunk 36]
```diff
@@ -24,14 +25,16 @@
...
-    String id;
+    String id, amnt;
     String name;
     RadioGroup radioGroup;
     LinearLayout totalLayout;
     int flag;
     Spinner poojaType;
     CheckBox paid;
     EditText moneyDonated;
+    EditText custompuja;
```
Comment (immadisairaj) [inline TempleApp/src/main/java/example/TempleApp/activities/UpdateData.java]: same as above

File: TempleApp/src/main/java/example/TempleApp/activities/ReadSingleData.java [hunk 37]
```diff
@@ -101,27 +101,27 @@ protected void onPreExecute() {
...
 
             try {
                 /*
                   Check Whether Its NULL???
                  */
                 if (jsonObject != null) {
 
-                    JSONObject user = jsonObject.getJSONObject(getString(R.string.user));
-                    name = user.getString(getString(R.string.name_));
+                    JSONObject user = jsonObject.getJSONObject(getString(example.TempleApp.R.string.user));
```
Comment (immadisairaj) [inline TempleApp/src/main/java/example/TempleApp/activities/ReadSingleData.java]: all the `example.TempleApp.R.string` are not necessary. `R.string` can be used.

revert all these kind

File: TempleApp/src/main/res/values/strings.xml [hunk 38]
```diff
@@ -56,14 +56,18 @@
     <string name="update_registered_pooja_details">Update Registered Pooja Details</string>
     <string name="update_the_details_of_donated_money">Update The Details Of Donated Money</string>
     <string name="update">Update</string>
-
-
+    <string name="fetching">Fetching Your Values</string>
+    <string name="Json_obj">Json obj</string>
+    <string name="Name">name</string>
+    <string name="amt">Amount:    Rs.</string>
+    <string name="paidstatus">Paid Status: </string>
```
Comment (immadisairaj) [inline TempleApp/src/main/res/values/strings.xml]: all these are already there in the resources. Why adding new ones?

**Comment by immadisairaj** [comment]
@gopikaharikumar You can send a new PR with the changes to the branch development.
I am closing this PR.

**Pull request closed without merging**